contrib/connectors/pandas/views.py

```diff
@@ -0,0 +1,270 @@
+            "(https://github.com/d3/d3-format/blob/master/README.md#format). "
+            "For instance, this default formatting applies in the Table "
+            "visualization and allow for different metric to use different "
+            "formats", True
+        ),
+    }
+    add_columns = edit_columns
+    page_size = 500
+    label_columns = {
+        'metric_name': _("Metric"),
+        'description': _("Description"),
+        'verbose_name': _("Verbose Name"),
+        'metric_type': _("Type"),
+        'source': _("Pandas Source Columns"),
+        'expression': _("Pandas Expression"),
+        'datasource': _("Datasource"),
+        'd3format': _("D3 Format"),
+        'is_restricted': _('Is Restricted')
+    }
+
+    def post_add(self, metric):
+        if metric.is_restricted:
+            security.merge_perm(sm, 'metric_access', metric.get_perm())
+
+    def post_update(self, metric):
+        if metric.is_restricted:
+            security.merge_perm(sm, 'metric_access', metric.get_perm())
+
+
+appbuilder.add_view_no_menu(PandasMetricInlineView)
+
+
+class PandasDatasourceModelView(DatasourceModelView, DeleteMixin):  # noqa
+    datamodel = SQLAInterface(models.PandasDatasource)
+
+    list_title = _('List Pandas Datasources')
+    show_title = _('Show Pandas Datasource')
+    add_title = _('Add Pandas Datasource')
+    edit_title = _('Edit Pandas Datasource')
+
+    list_columns = [
+        'link', 'changed_by_', 'modified']
+    order_columns = [
+        'link', 'changed_on_']
+    add_columns = ['name', 'source_url', 'format']
+    edit_columns = [
+        'name', 'source_url', 'format',
+        'filter_select_enabled', 'slices',
+        'fetch_values_predicate',
+        'description', 'owner',
+        'main_dttm_col', 'default_endpoint', 'offset', 'cache_timeout']
+    show_columns = edit_columns + ['perm']
+    related_views = [PandasColumnInlineView, PandasMetricInlineView]
+    base_order = ('changed_on', 'desc')
+    search_columns = (
+        'owner', 'name', 'source_url'
+    )
+    description_columns = {
+        'slices': _(
+            "The list of slices associated with this datasource. By "
+            "altering this datasource, you may change how these associated "
+            "slices behave. "
+            "Also note that slices need to point to a datasource, so "
+            "this form will fail at saving if removing slices from a "
+            "datasource. If you want to change the datasource for a slice, "
+            "overwrite the slice from the 'explore view'"),
+        'offset': _("Timezone offset (in hours) for this datasource"),
+        'name': _(
+            "The name of this datasource"),
+        'source_url': _(
+            "The URL used to access the raw data"),
+        'format': _(
+            "The format of the raw data, e.g. csv"),
+        'additional_parameters': _(
+            "A JSON-formatted dictionary of additional parameters "
```

Comment:
Could we have add something like: "These are the actual parameters for the read_* functions, see https://pandas.pydata.org/pandas-docs/stable/api.html"

Reply:
Done